strategies/iCARL/experiment.py

```diff
@@ -0,0 +1,191 @@
+    padded = np.pad(img, ((0, 0), (4, 4), (4, 4)), mode='constant')
+    random_cropped = np.zeros(img.shape, dtype=np.float32)
+    crop = np.random.randint(0, high=8 + 1, size=(2,))
+
+    # Cropping and possible flipping
+    if np.random.randint(2) > 0:
+        random_cropped[:, :, :] = \
+            padded[:, crop[0]:(crop[0]+32), crop[1]:(crop[1]+32)]
+    else:
+        random_cropped[:, :, :] = \
+            padded[:, crop[0]:(crop[0]+32), crop[1]:(crop[1]+32)][:, :, ::-1]
+    t = torch.tensor(random_cropped)
+    return t
+
+
+class Config(dict):
+    def __getattribute__(self, key):
+        try:
+            return self[key]
+        except KeyError:
+            raise AttributeError(key)
+
+    def __setattr__(self, key, value):
+        self[key] = value
+
+
+class iCARL(unittest.TestCase):
+    """
+        Reproducing iCaRL experiments from paper
+        "iCaRL: Incremental Classifier and Representation Learning",
+        Sylvestre-Alvise Rebuffi, Alexander Kolesnikov, Georg Sperl, Christoph H. Lampert;
+        Proceedings of the IEEE Conference on
+        Computer Vision and Pattern Recognition (CVPR), 2017, pp. 2001-2010
+        https://openaccess.thecvf.com/content_cvpr_2017/html/Rebuffi_iCaRL_Incremental_Classifier_CVPR_2017_paper.html
+    """
+
+    def test_iCIFAR100_batch10(self, override_args=None):
+        """
+            iCIFAR-100 with 10 batches - benchmark
+        """
+        # config
+        config = Config()
+        config.batch_size = 128
+        config.nb_exp = 10
+        config.memory_size = 2000  # K
+        config.epochs = 70
+        config.lr_base = 2.
+        config.lr_milestones = [49, 63]
+        config.lr_factor = 5.
+        config.wght_decay = 0.00001
+        # class incremental learning: classes mutual exclusive
+        config.fixed_class_order = [87, 0, 52, 58, 44, 91, 68, 97, 51, 15,
+                                    94, 92, 10, 72, 49, 78, 61, 14, 8, 86,
+                                    84, 96, 18, 24, 32, 45, 88, 11, 4, 67,
+                                    69, 66, 77, 47, 79, 93, 29, 50, 57, 83,
+                                    17, 81, 41, 12, 37, 59, 25, 20, 80, 73,
+                                    1, 28, 6, 46, 62, 82, 53, 9, 31, 75,
+                                    38, 63, 33, 74, 27, 22, 36, 3, 16, 21,
+                                    60, 19, 70, 90, 89, 43, 5, 42, 65, 76,
+                                    40, 30, 23, 85, 2, 95, 56, 48, 71, 64,
+                                    98, 13, 99, 7, 34, 55, 54, 26, 35, 39]
+        config.seed = 2222
+
+        #
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        torch.manual_seed(config.seed)
+        torch.cuda.manual_seed(config.seed)
+        np.random.seed(config.seed)
+        random.seed(config.seed)
+        torch.backends.cudnn.enabled = False
+        torch.backends.cudnn.deterministic = True
+
+        per_pixel_mean = get_dataset_per_pixel_mean(
```

Comment:
you can easily create the SplitCIFAR100 with avalanche using the `SplitCIFAR100` class directly. You can provide transformations and all the needed parameters (you can also fix the class order).

Reply:
yep, now it is better!